lib/uvobot/uvo_scraper.rb

```diff
@@ -0,0 +1,38 @@
+require 'curb'
+require_relative './uvo_parser'
+
+module Uvobot
+  class UvoScraper
+    SEARCH_URL = 'https://www2.uvo.gov.sk/evestnik?p_p_id=evestnik_WAR_eVestnikPortlets&p_p_lifecycle=1' \
+                 '&p_p_state=normal&p_p_mode=view&p_p_col_id=column-2&p_p_col_pos=1&p_p_col_count=2'.freeze
+    NEW_ISSUE_URL = 'https://www2.uvo.gov.sk/evestnik/-/vestnik/aktual'.freeze
+    IT_CONTRACTS_CODE = 72
+
+    def initialize(parser = Uvobot::UvoParser, html_client = Curl)
+      @parser = parser
+      @html_client = html_client
+    end
+
+    def issue_ready?(release_date)
+      html = @html_client.get(NEW_ISSUE_URL).body
+      header = @parser.parse_issue_header(html)
+      identifier = release_date.strftime('/%Y - %-d.%-m.%Y')
+      header.include?(identifier)
+    end
+
+    def get_announcements(release_date)
+      date = release_date.strftime('%d.%m.%Y')
+      search_query = { cpv: IT_CONTRACTS_CODE, datumZverejneniaOd: date, datumZverejneniaDo: date }
+      html = @html_client.post(SEARCH_URL, search_query).body
+
+      [@parser.parse_page_info(html), @parser.parse_announcements(html)]
+    end
+
+    def get_announcement_detail(url)
+      html = @html_client.get(url).body
+      @parser.parse_detail(html)
+    rescue StandardError => e
+      { amount: 'Parsovanie zlyhalo.' }
```

Comment:
Skor mi vadi to, ze sa `amount` pouziva na error handling. nech to vrati nil ak sa to nepodarilo a hore sa rozhodne co to znamena. Imho by to malo do fora poslat post, ale napisat ze sa detaily nepodarilo extrahovat. Nechat tam len link. Hm?